Setup APP

Rushyanth opened the pull request.

Commit:
Add dummy file

Commit:
Add app setup and templates

Commit:
Add requirements

Commit:
Modify Templates

Commit:
Modify __init__.py

Add SQLalchemy
Add Migrate
Add Login Manager

Commit:
Add classes for
Register Login forms

Commit:
Add the database classes
Users class
TaskList Class

Commit:
Add Migrations for database

Commit:
Routes for all the functions
Edit Task
Delete Task
Add Task
Logout

adityav-verma commented on app/models.py:
```diff
@@ -0,0 +1,46 @@
+	def is_own_task(self, id):
+		if id == current_user.id :
+			return True
+		else:
```
No need of this `else`

Rushyanth replied:
done

adityav-verma commented on app/models.py:
```diff
@@ -0,0 +1,46 @@
+		else:
+			return False
+	 
+@login.user_loader
```
Move this function to `app/__init__.py`. This function does not make any sense on `models.py` file.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+def home():
+	user = {'username': 'rushi'}
+	return render_template('home.html')
+
```
What is this route and function???? Remove it.

Rushyanth replied:
Its the home page
removed this line
user = {'username': 'rushi'}

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		user.set_password(password)
+		db.session.add(user)
+		db.session.commit()
+
```
Let's add `try/except` block wherever we do DB operations.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		if user is None or not user.check_password(password):
+			flash("User doesnt exist or password is incorrect")
+			return redirect(url_for('login_get'))
+		else:
```
`else` not needed.

Rushyanth replied:
done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+			return redirect(url_for('login_get'))
+		else:
+			login_user(user)								# Authentication of User
+			session['user_id'] = user.id
```
Why adding this to the `session`? Remove it if not needed.

Rushyanth replied:
Done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	# Get the tasks to homepage
+	# Retreive data
+	tasks = current_user.tasks.all()
+	print(tasks)
```
Remove `print`, shouldn't be there in production code.

Rushyanth replied:
done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		response.headers['Expires'] = '-1'
+		return response
+	# Close cursor
+	cur.close()
```
What is this `cur.close()`? Remove it if not needed. Also remove the related imports.

Rushyanth replied:
done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		response.headers['Pragma'] = 'no-cache'
+		response.headers['Expires'] = '-1'
+		return response
+	else:
```
Why is this `else` needed? You are doing the exact same thing as above. Handle it in same code.

Rushyanth replied:
done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		task = TaskList(title=title, body=body, status=status, user_id=current_user.id)
+		db.session.add(task)
+		db.session.commit()
+		
```
Let's add `try/except` block around db operations.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+		# Get Task Details
+		title = form.title.data
+		body = form.body.data
+		status = False
```
Why is this `status` `False`?

Rushyanth replied:
Initial status of all the newly created tasks is False

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	
+	if task is not None and task.is_own_task(task.user_id):
+		return render_template('editTask.html', task=task)
+	else :
```
Remove this extra `else`, not needed.

Rushyanth replied:
done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+def editTask_post(id):
+	# Get data
+	task = TaskList.query.filter_by(id=id).first()
+	
```
Add `user_id` owner check to this endpoint too.

adityav-verma replied:
And also you should have an EditTask form, and then do `form.validate()`.

Rushyanth replied:
owner check is not needed here.
I didnot use WTF form for Edit Task.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	task.status = False
+
+	db.session.add(task)
+	db.session.commit()
```
`try/except`

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+@app.route('/deleteTask/<string:id>', methods=['POST'])
+@login_required
+def deleteTask(id):
+	task = TaskList.query.filter_by(id=id).first()
```
Add `user_id` owner check here. Any logged in user would be able to delete any task.

Rushyanth replied:
Not necessary. Another user canot go to that page by typing the link
It automatically redirects.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+def deleteTask(id):
+	task = TaskList.query.filter_by(id=id).first()
+	db.session.delete(task)
+	db.session.commit()
```
`try/except`

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	if status == 'False':
+		status = True
+	else:
+		status = False
```
What are you trying to do here? Please explain.

Rushyanth replied:
Changing the status of the task as done or not Done

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	else:
+		status = False
+	
+	task = TaskList.query.filter_by(id=id).first()
```
`user_id` owner check.

Rushyanth replied:
Not necessary. It Automatically redirects on button click.

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+	task = TaskList.query.filter_by(id=id).first()
+	task.status = status
+	db.session.add(task)
+	db.session.commit()
```
`try/except`

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+@app.route('/logout')
+@login_required
+def logout():
+	session.clear();
```
Use `Flask-Login` `logout_user()`

Rushyanth replied:
Done

adityav-verma commented on config.py.sample:
```diff
@@ -0,0 +1,4 @@
+import os
+
+class Config(object):
+    SECRET_KEY = os.environ.get('SECRET_KEY') or 'dummy'
```
Add all the configs with default values in the sample. I am sure you added more configurations like `SQLAlchemy` etc.

Rushyanth replied:
Done

coderadi commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+def addTask_post():
+	form = AddTaskForm(request.form)
+	if form.validate():
+		form = AddTaskForm(request.form)	
```
Where is `form.validate()`?

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,218 @@
+def addTask_post():
+	form = AddTaskForm(request.form)
+	if form.validate():
+		form = AddTaskForm(request.form)	
```
Remove this `form = AddTaskForm(request.form)`, you have already done this before validate function.

Rushyanth replied:
Done

adityav-verma commented:
Also add a proper PR description.
## What?
What does the PR do?

## Why?
Why is this PR needed?

Commit:
Made the necessary changes
Removal of unnecessary if else
Addition of user-id check
try/except yet to be done

Commit:
All the necessary configurations

Commit:
Added permission for Edit task and Delete Task

adityav-verma commented on app/routes.py:
```diff
@@ -0,0 +1,235 @@
+			print("in databaseerror exception block")
+			flash("Could not register", "danger")
+			return redirect(url_for('register_get'))
+		else:
```
Why this `else`?

Commit:
Format PEP8 errors

Commit:
PEP8 corrections

Commit:
Change of PEP8 errors

Commit:
Change in styling

Commit:
Refactor and cleanup
